### PR TITLE
TMA 710: segments filter in user and user filters bricks

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 1]
```diff
@@ -167,18 +167,18 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            if params.segments_filter
-              segment_uris = params.segments_filter.map(&:uri)
-              domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
+            if params.segments
+              segment_uris = params.segments.map(&:uri)
+              domain_clients = domain_clients.select { |c| segment_uris.include?(c.segment_uri) }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Has the variable name 'c' (UncommunicativeVariableName)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 2]
```diff
@@ -167,18 +167,18 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            if params.segments_filter
-              segment_uris = params.segments_filter.map(&:uri)
-              domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
+            if params.segments
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'params.segments' 4 times (DuplicateMethodCall)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 3]
```diff
@@ -168,17 +168,17 @@ def call(params)
...
                         domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
                       end
                       working_client_ids = []
                       res = []
                       res += new_users.group_by { |u| u[:pid] }.flat_map do |client_id, users|
                         fail "Client id cannot be empty" if client_id.blank?
                         c = domain.clients(client_id, data_product)
-                        if params.segments_filter && !segment_uris.include?(c.segment_uri)
-                          puts "Client #{client_id} is outside segments_filter #{params.segments_filter}"
+                        if params.segments && !segment_uris.include?(c.segment_uri)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 4]
```diff
@@ -168,17 +168,17 @@ def call(params)
                       end
                     when 'sync_domain_client_workspaces'
                       domain_clients = domain.clients(:all, data_product)
-                      if params.segments_filter
-                        segment_uris = params.segments_filter.map(&:uri)
+                      if params.segments
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'params.segments' 4 times (DuplicateMethodCall)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 5]
```diff
@@ -271,9 +271,14 @@ def self.maqlify_filters(filters, project_users, options = {})
...
         expression, errors = create_expression(f, labels_cache, lookups_cache, attrs_cache, options)
         profiles_uri = if options[:type] == :muf
+<<<<<<< HEAD
                          project_user = project_users.find { |u| u.login == login }
 
                          project_user.nil? ? ('/gdc/account/profile/' + login) : project_user.profile_url
+=======
+                         uri = project_users.find { |u| u.login == login }.profile_url
+                         uri.nil? ? ('/gdc/account/profile/' + login) : uri
+>>>>>>> TMA-638: Fix removing user with MUFs
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has unexpected token tcolon (Syntax)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has unexpected token trshft (Syntax)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: unexpected token tRSHFT
(Using Ruby 2.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: unexpected token tCOLON
(Using Ruby 2.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 6]
```diff
@@ -271,9 +271,14 @@ def self.maqlify_filters(filters, project_users, options = {})
       create_filter_proc = proc do |login, f|
         expression, errors = create_expression(f, labels_cache, lookups_cache, attrs_cache, options)
         profiles_uri = if options[:type] == :muf
+<<<<<<< HEAD
                          project_user = project_users.find { |u| u.login == login }
 
                          project_user.nil? ? ('/gdc/account/profile/' + login) : project_user.profile_url
+=======
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has unexpected token teqq (Syntax)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: unexpected token tEQQ
(Using Ruby 2.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 7]
```diff
@@ -271,9 +271,14 @@ def self.maqlify_filters(filters, project_users, options = {})
       create_filter_proc = proc do |login, f|
         expression, errors = create_expression(f, labels_cache, lookups_cache, attrs_cache, options)
         profiles_uri = if options[:type] == :muf
+<<<<<<< HEAD
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has unexpected token tlshft (Syntax)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: unexpected token tLSHFT
(Using Ruby 2.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/595/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/401/ : FAILURE in 51s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/709/ : FAILURE in 42s

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Pull request closed without merging**